MST/CMakeLists.txt

```diff
@@ -0,0 +1,215 @@
+################
+# Iman Jehanzeb
+################
+
+cmake_minimum_required(VERSION 3.18)
+
+cmake_host_system_information(RESULT Ncpu QUERY NUMBER_OF_PHYSICAL_CORES)
+message(STATUS "CMake ${CMAKE_VERSION} with max. ${Ncpu} threads")
+
+project(MST LANGUAGES Fortran C CXX CUDA)
```

Comment:
At the moment you only need Fortrn and C. CXX is normally not needed for MST. Also make CUDA optional, since not every PC has GPU for calculations.